src/guiapi/src/window_icon.cpp

```diff
@@ -9,35 +11,91 @@ void window_icon_init(window_icon_t *window) {
 }
 
 void window_icon_draw(window_icon_t *window) {
-    if (window->flg & WINDOW_FLG_INVALID) {
+    if (window->IsInvalid()) {
         //point_ui16_t pt = {window->rect.x, window->rect.y};
-        //display::DrawIcon(pt, window->id_res, window->color_back, (window->flg & WINDOW_FLG_FOCUSED)?ROPFN_SWAPBW:0);
+        //display::DrawIcon(pt, window->id_res, window->color_back, (window->IsFocused())?ROPFN_SWAPBW:0);
         uint8_t ropfn = 0;
-        if ((window->flg & WINDOW_FLG_DISABLED)) { // that could not be set, but what if
+        if ((window->IsBWSwapped())) { // that could not be set, but what if
             ropfn |= ROPFN_DISABLE;
         }
-        if ((window->flg & WINDOW_FLG_FOCUSED)) {
+        if ((window->IsFocused())) {
             ropfn |= ROPFN_SWAPBW;
         }
 
         render_icon_align(window->rect, window->id_res, window->color_back,
             RENDER_FLG(window->alignment, ropfn));
-        window->flg &= ~WINDOW_FLG_INVALID;
+        window->Validate();
+        ;
     }
 }
 
-const window_class_icon_t window_class_icon = {
-    {
-        WINDOW_CLS_ICON,
-        sizeof(window_icon_t),
-        (window_init_t *)window_icon_init,
-        0,
-        (window_draw_t *)window_icon_draw,
-        0,
-    },
-};
-
 void window_icon_t::SetIdRes(int16_t id) {
     id_res = id;
     Invalidate();
 }
+
+window_icon_t::window_icon_t(window_t *parent, rect_ui16_t rect, uint16_t id_res, is_closed_on_click_t close)
+    : window_t(parent, rect, is_dialog_t::no, close)
+    , id_res(id_res)
+    , alignment(ALIGN_CENTER) {
+}
+//Icon rect is increased by badding, than icon is centered inside it
```

Comment:
Icon rect is increased by padding, than icon is centered inside it
```suggestion
//Icon rect is increased by padding, icon is centered inside it
```